### PR TITLE
Adds digitigrade-accessible footwraps

### PR DESCRIPTION
**Pull request by Anvilman6**
Adds infrastructure for digitigrade-usable shoes, able to be added to in the future. Adds 3 types of footwraps. 2 are currently active, one for common crew and one for Security, and both will appear on relevant jobs at roundstart; the 3rd type shouldn't appear as of yet as conditions are not in place for it. Both active versions can be worn by digitigrade or straight leg races, and there are flags to set existing or later new wraps as either straight or digitigrade leg locks as well as a hybrid of both. Shoes will change display state according to if a uniform is enforcing straight leg side view on digitigrades or not, or if the person is lacking a uniform+suit.

A huge thank you to @SomeguyManperson and @Mqiib as you guys really made this hellacious project possible and prevented me from tearing my face off due to shoddy mobcode and itemcode. And also thank you to @redmoogle for helping me realize that the errors I was getting on compile weren't from the code, but were from having Dream Daemon open and running in the background - I am quite stupid.

# General Documentation

### Intent of your Pull Request

Adding a shoe type that is wearable by digitigrade crew (Lizards with alternative legs, Polysmorphs in all forms). With the right item flags they can also be used by straight legged crew (which the first Common and Security wraps possess).

### Why is this change good for the game?

Provides more customization options for digitigrade players, and for other players if they want to wear them too.

# Wiki Documentation

### Briefly describe your PR and the impacts of it, in layman's terms. 

TODO: add entries related to the footwraps on the Wiki and clothing lists, adjust information saying there's no digitigrade-accessible footwear.

### What should players be aware of when it comes to the changes your PR is implementing?

You will start with either Common Footwraps or Reinforced Footwraps, depending on if you have a normal job or a Security job. Currently there is no way to obtain footwraps outside of roundstart/latejoin start / admin spawning, but I will look into adding them to vending machines soon and/or having them be craftable in some form.

### What general grouping does this PR fall under? 
Clothing, Jobs

# Changelog

:cl:  
rscadd: Adds footwraps and related infrastructure to the game!
wip: Wraps should be functional at this stage, but there are a few modular additions I will be looking at later. Such as adding them to vendors, potentially adding sound effects to them, or potentially adding other types of wraps beyond Common/Security. Suggestions are welcome here.
tweak: Added more infrastructure to footwear. Shoes can be flagged as digitigrade usable or not for easy modular additions.
tweak: Creates more aggressive cases of item state updating and checking for shoes, so be advised if designing shoes with alternative view states.
imageadd: Item sprites, wrap on-character sprites, and situational side view sprites! I am not a spriter, so if people think these need touchups, I would welcome spriter input and help for a patch.
/:cl:

### CODE REVIEW

File: code/modules/jobs/job_types/_job.dm [hunk 1]
```diff
@@ -215,6 +215,8 @@
 	var/duffelbag = /obj/item/storage/backpack/duffelbag
 
 	var/pda_slot = SLOT_BELT
+	var alt_shoes = /obj/item/clothing/shoes/xeno_wraps // Default digitgrade shoes assignment variable
```
Comment (SomeguyManperson) [inline code/modules/jobs/job_types/_job.dm]:
```suggestion
	var/alt_shoes = /obj/item/clothing/shoes/xeno_wraps // Default digitgrade shoes assignment variable
```

File: code/modules/jobs/job_types/_job.dm [hunk 2]
```diff
@@ -215,6 +215,8 @@
...
+	var alt_shoes_s = /obj/item/clothing/shoes/xeno_wraps/jackboots // Digitigrade shoes for Sec assignment variable
```
Comment (SomeguyManperson) [inline code/modules/jobs/job_types/_job.dm]:
```suggestion
	var/alt_shoes_s = /obj/item/clothing/shoes/xeno_wraps/jackboots // Digitigrade shoes for Sec assignment variable
```

**Comment by UselessTheremin** [comment]
fetish content

good pr

**Comment by Anvilman6** [comment]
> fetish content
> 
> good pr

Now draw her putting footwraps on

**Comment by ToasterBiome** [comment]
why tho

**Comment by Mqiib** [comment]
This was way more a pain in the ass than it should have been, glad you were able to finish it!

**Comment by boodaliboo** [comment]
lizards dont have many downsides i dont think this should be a thing to compensate for there already increased melee damage

**Comment by Anvilman6** [comment]
> lizards dont have many downsides i dont think this should be a thing to compensate for there already increased melee damage

@boodaliboo They do not have increased melee damage, they have visually / audibly different melees, but it's the same baseline damage. Moths also use the same visuals / audio and deal the same damage as everything else.

https://github.com/yogstation13/Yogstation/blob/c2f0f7f380a680ba29455809ed942d9b443d17f0/code/modules/mob/living/carbon/human/species_types/lizardpeople.dm#L16-L18

https://github.com/yogstation13/Yogstation/blob/c2f0f7f380a680ba29455809ed942d9b443d17f0/code/modules/mob/living/carbon/human/species_types/mothmen.dm#L10-12

**Comment by Mqiib** [comment]
> lizards dont have many downsides i dont think this should be a thing to compensate for there already increased melee damage

Anvil is correct, lizards do NOT have more damage.  On top of that, lizards can already wear shoes if they pick the normal leg option, so all of this "it's a buff to lizards" is a completely moot point. If lizard players didn't want to play with the obvious downside of not being able to wear shoes then they could just choose not to anyways. This is almost totally an aesthetic change. It's only a "buff" to polysmorphs because they are forced to have digitigrade legs, but polysmorphs are comparatively weak already since they have increased burn and heat mods.

**Comment by Marmio64** [comment]
only if you also make lizards take damage from hearing the L word

**Comment by CrazyDrakkon** [comment]
> 
> 
> only if you also make lizards take damage from hearing the L word

only if you also make the one saying it take toxic damage

**Review by nesisknesis — APPROVED** [review]

**Comment by MegaEmpirical** [comment]
I have made a mock up in support of Plasmaman HoS':

image
